packages/react-router/docs/guides/migrating.md

```diff
@@ -140,4 +141,25 @@ const App = () => (
     <Route path='/contact' component={Contact} />
   </Switch>
 )
+
+```
+
+### `<Redirect>`
+
+In v3, if you wanted to redirect from one path to another, for instance /welcome to /, you would use `<IndexRedirect >`.
+
+```js
+// v3
+<Route path="/" component={App}>
+  <IndexRedirect to="/welcome" />
+</Route>
+
+```
+
+In v4, you can achieve the same functionality using `<Redirect>`.
+
+```js
+// v4
+<Route path="/" component={() => <Redirect to="/welcome" component={App} />} />
```

Comment:
A few tweaks should be made here.

1. This `<Route>` needs the `exact` prop. Otherwise, `path='/'` will match every route, so everything would be redirected.
2. This should be using the `render` prop instead of `component`. Using `component` with inline functions causes unnecessary unmounts and mounts. Technically here that doesn't matter because we just redirect, but we should do our best to avoid any example code that uses `component` with inline functions.